packages/context/src/__tests__/acceptance/binding-config.feature.md

```diff
@@ -0,0 +1,84 @@
+# Feature: Context bindings - injecting configuration for bound artifacts
+
+- In order to receive configuration for a given binding
+- As a developer
+- I want to set up configuration for a binding key
+- So that configuration can be injected by the IoC framework
+
+# Scenario: configure an artifact before it's bound
+
+- Given a `Context`
+- Given a class RestServer with a constructor accepting a single argument
+  `config: RestServerConfig`
+- Given `RestServer` ctor argument is decorated with `@config()`
+- When I bind a configuration object `{port: 3000}` to
+  `servers.rest.server1:$config`
+- And bind the rest server to `servers.rest.server1`
+- And resolve the binding for `servers.rest.server1`
+- Then I get a new instance of `RestServer`
+- And the instance was created with `config` set to `{port: 3000}`
+
+```ts
+class RestServer {
+  constructor(@config() public config: RestServerConfig) {}
+}
+
+const ctx = new Context();
+
+// Bind configuration
+ctx.configure('servers.rest.server1').to({port: 3000});
+
+// Bind RestServer
+ctx.bind('servers.rest.server1').toClass(RestServer);
+
+// Resolve an instance of RestServer
+// Expect server1.config to be `{port: 3000}
+const server1 = await ctx.get('servers.rest.server1');
+```
+
+# Scenario: configure an artifact with a dynamic source
+
+- Given a `Context`
+- Given a class RestServer with a constructor accepting a single argument
+  `config: RestServerConfig`
+- Given `RestServer` ctor argument is decorated with `@config()`
+- When I bind a configuration factory of `{port: 3000}` to
+  `servers.rest.server1:$config`
```

Comment:
Is it `servers.rest.server1.$config` here instead of `:`?

Reply:
`servers.rest.server1:$config` is the binding key for config.

Reply:
It's written as `.$config` from https://github.com/strongloop/loopback-next/pull/2259/files/28db47cd98ed35288764fd47c33b8c806e87cd3c#diff-3ecbb2bbc45d751a5c7190943c36f277R570.